### PR TITLE
[SPARK-37834][PYTHON][INFRA] Reenable length check in Python linter

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR reenables length check in Python linter (E501 in flake8) that is disabled from https://github.com/apache/spark/pull/34297

### Why are the changes needed?

To have the consistent style, and follow the convention.

### Does this PR introduce _any_ user-facing change?

No, dev-only.

### How was this patch tested?

Existing unitttest should cover all.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @dongjoon-hyun @zero323 @xinrong-databricks @itholic @ueshin @nchammas FYI

**Comment by HyukjinKwon** [comment]
cc @dchvn too fyi

File: python/pyspark/resource/profile.py
Comment (HyukjinKwon) [inline python/pyspark/resource/profile.py:158]: this is consistent with another place in this file. I think it was just a mistake.

File: python/pyspark/sql/session.py
Comment (HyukjinKwon) [inline python/pyspark/sql/session.py:671]: MyPy can understand `_jvm` assert so it doesn't require `# type: ignore[attr-defined]`.

File: python/pyspark/resource/profile.py
Comment (dongjoon-hyun) [reply python/pyspark/resource/profile.py:158]: Yes, it looks like that.

File: python/pyspark/resource/requests.py
Comment (dongjoon-hyun) [inline python/pyspark/resource/requests.py:309]: This file changes also look reasonable.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @HyukjinKwon !

**Review by itholic — APPROVED** [review]
LGTM once tests pass

**Comment by zero323** [comment]
> This PR reenables length check in Python linter (E501 in flake8). This was mistakenly removed from #34655.

Actually, I disabled E501 in https://github.com/apache/spark/pull/34297, mostly following [line length notes](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html?highlight=E501#line-length) from the black documentation. Specifically

> However, sometimes it won’t be able to without breaking other rules. In those rare cases, auto-formatted code will exceed your allotted limit.

In other cases `line-length` in black config should be respected and kind of duplicate what we have in style config.

That being said, LGTM.

**Comment by HyukjinKwon** [comment]
Ah, thanks for clarification @zero323. If this bugs us with black formatter, let's think about switching it back in the future 👍

**Review by zero323 — APPROVED** [review]

**Comment by zero323** [comment]
> If this bugs us with black formatter, let's think about switching off back in the future

The most likely violators are `type: ignore` comments and number of these is going down anyway.

Most of the remaining ones can be handled with magic comma (for example, I don't think we'll need ignore on `ml.param.shared` once #34674 is merged).

File: python/pyspark/tests/test_readwrite.py
Comment (zero323) [inline python/pyspark/tests/test_readwrite.py:236]: I don't feel strongly about it, but brackets should work in such cases, without introducing new variables

```python
    "mapreduce.job.outputformat.class": (
        "org.apache.hadoop.mapreduce.lib.output.SequenceFileOutputFormat"
    ),
```

Comment (HyukjinKwon) [reply python/pyspark/tests/test_readwrite.py:236]: im working on another PR to enable some rules - I will create the Pr today. Let me fix it in the PR together because that Pr touches here and there too..

**Comment by HyukjinKwon** [comment]
Tests passed.

Merged to master.

**Pull request closed without merging**
